modules/fileout_netcdf/FONcArrayStructure.cc

```diff
@@ -0,0 +1,220 @@
+        throw BESInternalError(s, __FILE__, __LINE__);
+    }
+
+    if (false == isNetCDF4_ENHANCED()) {
+        throw BESInternalError("Fileout netcdf: We don't support array of structure for the classical model now. ",
+                               __FILE__, __LINE__);
+    }
+
+    Constructor::Vars_iter vi = as_v->var_begin();
+    Constructor::Vars_iter ve = as_v->var_end();
+    for (; vi != ve; vi++) {
+        BaseType *bt = *vi;
+        if (bt->send_p()) {
+            BESDEBUG("fonc", "FONcArrayStructure::convert - converting " << bt->name() << endl);
+
+            auto fsf = new FONcArrayStructureField(bt, _as);
+            _vars.push_back(fsf);
+            fsf->convert(embed,_dap4,is_dap4_group);
+        }
+    }
+
+}
+
+/** @brief Define the members of the array of structure in the netcdf file
+ *
+ * We still follow the current structure implementation to define the members of
+ * the structure to include the name of the structure in their name.
+ *
+ * @note This will call the FONcArrayStructureField's define() method for individual member.
+ * Because the FONcArrayStructure::convert() method above only
+ * builds a FONcArrayStructure for elements of the DAP Structure with send_p true,
+ * the code is protected from trying to operate on DAP variables that should
+ * not be sent. This is important because those variables likely don't contain
+ * any values!
+ *
+ * @param ncid The id of the netcdf file
+ * @throws BESInternalError if there is a problem defining the structure
+ */
+void FONcArrayStructure::define(int ncid)
+{
+
+    if (!d_defined) {
+        BESDEBUG("fonc", "FONcArrayStructure::define - defining " << d_varname << endl);
+        vector<FONcArrayStructureField *>::const_iterator i = _vars.begin();
+        vector<FONcArrayStructureField *>::const_iterator e = _vars.end();
+        for (; i != e; i++) {
+            FONcArrayStructureField *fbt = (*i);
+            BESDEBUG("fonc", "defining " << fbt->name() << endl);
+            fbt->define(ncid);
+        }
+
+        d_defined = true;
+
+        BESDEBUG("fonc", "FONcArrayStructure::define - done defining " << d_varname << endl);
+    }
+
+}
+
+/** @brief write the member variables of the structure to the netcdf
+ * file
+ *
+ * @param ncid The id of the netcdf file
+ * @throws BESInternalError if there is a problem writing out the
+ * members of the structure.
+ */
+void FONcArrayStructure::write(int ncid)
+{
+
+    BESDEBUG("fonc", "FONcArrayStructure::write - writing " << d_varname << endl);
+    if (d_is_dap4)
+        _as->intern_data();
+
+    for (const auto &var:_vars) {
+        var->write(ncid);
+    }
+    BESDEBUG("fonc", "FONcArrayStructure::define - done writing " << d_varname << endl);
+}
+
+/** @brief Returns the name of the structure
+ *
+ * @returns The name of the structure
+ */
+string FONcArrayStructure::name()
```

Comment:
This method can be const

Reply:
This is virtual method that inherits from FONcBase.h. I need to change a lot of header files to make it compile. So skip it until we want to update the whole module.